### PR TITLE
Try cyvcf

### PR DESCRIPTION
**Pull request by moonso**
Changed so we use cyvcf2 in vcf adapter

### CODE REVIEW

**Comment by moonso** [comment]
Not ready yet...

**Comment by moonso** [comment]
Ping @robinandeer maybe we should merge this first and then add the region filter in another PR?

**Comment by moonso** [comment]
Ok boys @robinandeer , @guillermo-carrasco , @brainstorm we are ready for cyvcf! Please merge this when possible :walking:

**Comment by brainstorm** [comment]
Yay, good job Måns! :)

File: puzzle/models/variant.py [hunk 1]
```diff
@@ -158,14 +158,17 @@ def _set_variant_id(self, variant_id=None):
         """Set the variant id for this variant"""
         if not variant_id:
             variant_id = '_'.join([
-                self['CHROM'],
-                self['POS'],
-                self['REF'],
-                self['ALT']
+                self.CHROM,
+                str(self.POS),
```
Comment (brainstorm) [inline puzzle/models/variant.py]: Why is this type casting semantically right/needed (position should be an integer, or)?

Comment (moonso) [inline puzzle/models/variant.py]: Yes POS should be an integer but `join` only accepts strings

File: puzzle/plugins/vcf/mixins/variant_mixin.py [hunk 2]
```diff
@@ -369,78 +219,78 @@ def _format_variants(self, variant_line, index, case_obj, add_all_info=False):
...
-            info_line = variant_dict['INFO']
-        )
-
-        variant = Variant(
-            **{column: variant_dict.get(column, '.')
-                for column in self.variant_columns}
+        info_dict = dict(variant.INFO)
+        
+        variant_obj = Variant(
+                CHROM=variant.CHROM.lstrip('chrCHR'),
```
Comment (brentp) [inline puzzle/plugins/vcf/mixins/variant_mixin.py]: careful with this:
```
'hcrchh.aar'.lstrip('chr') == '.aar'
```

Comment (moonso) [inline puzzle/plugins/vcf/mixins/variant_mixin.py]: Thanks! I'll think of a solution

**Pull request merged**